### PR TITLE
Fishing 2 - Return of the Fish

### PR DESCRIPTION
**Pull request by ToasterBiome**
image

# Document the changes in your pull request

Fishing tiles have very basic functionality right now which is basically just a **fishable** component attached to a water tile that includes its the fishing datum. This can be expanded on with different biomes or different mechanics.
## Population
Fishing tiles can have a "population" variable that fluctuates over the round, typically regenerating to a maximum level of population. Every time you successfully catch something from the tile, the population in that tile decreases. If it ever reaches zero, it will not regenerate naturally for the rest of the round. This prevents autofishing and will make you actually move around.

### Chum
A chummer machine would turn meat dishes into **chum**. This chum can be used on a fishing tile to artificially increase the population in the tile up to a maximum. If you use special chum on an overfished tile (a population of zero) you will kickstart its natural regeneration.

### Chemicals
Using existing or new chemicals (maybe Saltpetre, Ammonia, etc.) can be used to increase the population regeneration up to a maximum point.

### Fish Finder
A new device like a health scanner or plant analyzer which can scan a tile to tell you various statistics about the tile like:
* Biome
* Population
* Max Population
* Regeneration Rate
* Maximum Regeneration Rate

Also able to be used to scan fishing clothing, rods, and bait to tell you their fishing power.

### Tackle Box
Enjoy a luxurious new storage item to keep all your fishing gear in! No longer will you be enslaved to only carrying 27 apprentice bait in your bag!

### Fishing Encyclopedia
A book in-game that shows you all fish, where to catch them, their size and weight ranges, and what they can be processed into. Some say there are more fish in the galaxy, but the book is never wrong...

## Fishing in things other than water tiles
An important part of a complex and deep fishing system in video games is overcoming the fact that in real life, fishing is really boring. Within a game world we can extend the concept of fishing to catch more interesting things in more interesting locations.

### Fishing Portal
Buildable machines that act as bluespace portals that you can fish from. They have a much lower regeneration rate, population, and generally weaker loot tables than fishing tiles. These provide an alternative, and most importantly, *portable* method of catching fish. Ice fishing in the AI satellite anyone?

Moving it around to certain areas on the station will change what fish come out of it!

## Cargo and Fishing
Cargo can play a big part in the progression of fishing over a round. Since Cargo Technicians are more than likely extremely bored, we can add cargo shuttle exclusive items that can push Cargo Techs to be part time stockers, full time fishers!

### Cargo Fishing Portal
A better, non-deconstructable version of a Fishing Portal, only orderable through the cargo console.

### Cargo Chum
A reorderable, and renewable source of chum (not that normal chum isn't renewable)!

### Fishing Bounties
Since our bounty system is pretty open, there's no harm in adding 1-2 fishing bounties a round that anyone can complete, or at least try to.

## Biomes and Loot
Biomes are an interesting way to diversify a loot pool and make it so you will actively seek out areas to fish. The **fishing_loot** datum can easily be modified and subtyped into different biomes like the one we currently have in game now:

| Rarity     | Loot |
| -------- | -------- |
| Junk     | Plate, Cola, Glass Shard |
| Common   | Goldfish, Salmon, Bass, Leech |
| Uncommon | Giant Goldfish, Shrimp, Pufferfish, Tuna|
| Rare     | Squid, Bluespace Crystal |

### Proposed Additional Loot Biomes
* Maintenance
    * Fishing in here will give you things more tailored to the maint-dwelling inhabitance, like cable coils, MREs, flashlights, etc. Rarely in here you will fish up the **Ratfish**, which will act as a pair of wirecutters!
* Perma
    * Fishing here will suck. Although you can be rarely rewarded with things like cable coils, most of your loot will be generic trash and soggy tofu.
* Lavaland
    * Yooo shit lava fishing! You will need a special fishing rod that can handle the extreme temperatures of the magma, and these pools don't naturally regenerate! Fishing here will net you more minerals than fish, along with some lavaland exclusives like a **Lava Eel**, **Bubblegum Guppy**, or the rare **Plasma Pirahna**!

### Of course I complete my objectives fast, I got fish to catch!
Everyone knows that bad guys cheat and that's the only reason they beat you. Using a special, one-use traitor item - Concentrated Lizard Chum - a Fishing Portal will turn it blood red and give access to a *special* biome instead, along with a passive boost to fishing power inside it. Functionally, its loot is the same as the standard "Water" biome with a few extra goodies mixed in like the **Energy Swordfish**, the chance to *extremely rarely* catch **telecrystals**, and the elusive **Black and Red Tacklebox**.

# Spriting
If you are adding new sprites to the game please add a picture of the sprite in its relative context, ie. Clothing on a mob.

See above.

Fish finder:
image

Tacklebox:
Aseprite_D7qVwtFjSl

# Wiki Documentation

We don't have a dedicated page for fishing (yet) but all this needs to be documented (whoops)

# Changelog

:cl:  
/:cl:

### CODE REVIEW

**Comment by VaelophisNyx** [comment]
poggers-fish-lol

**Review by cuackles — APPROVED** [review]
man i love fishes

File: _maps/_basemap.dm [hunk 1]
```diff
@@ -1,4 +1,4 @@
-//#define LOWMEMORYMODE //uncomment this to load centcom and runtime station and thats it.
+#define LOWMEMORYMODE //uncomment this to load centcom and runtime station and thats it.
```
Comment (redmoogle) [inline _maps/_basemap.dm]:
```suggestion
//#define LOWMEMORYMODE //uncomment this to load centcom and runtime station and thats it.
```

**Review by redmoogle — CHANGES_REQUESTED** [review]
Probably shouldnt include this change

File: tgui/packages/tgui/interfaces/FishingEncyclopedia.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import { useBackend } from '../backend';
+import { Box, Button, Section, LabeledList } from '../components';
+import { Window } from '../layouts';
+
+export const FishingEncyclopedia = (props, context) => {
+  const { act, data } = useBackend(context);
+
+  const fish_map = data.f_list.map(fish => {
+    return (
+      <Box>{fish.name}{fish.min_weight}</Box>
```
Comment (redmoogle) [inline tgui/packages/tgui/interfaces/FishingEncyclopedia.js]:
```suggestion
      <Box key={fish.name}>{fish.name}{fish.min_weight}</Box>
```

**Comment by Bibby0110** [comment]
God I love fishing.
God I hate stale PRs :(

**Comment by ToasterBiome** [comment]
will reopen once we have elements (need them for lavaland fishing)

**Pull request closed without merging**